substrate/frame/support/procedural/src/deprecation.rs

```diff
@@ -97,15 +98,34 @@ pub fn get_deprecation_enum<'a>(
 			.unwrap_or_else(|| Ok(None))
 	}
 
-	let children = attrs
+	let parent_deprecation = parse_deprecation(path, parent_attrs)?;
+
+	let children = children_attrs
 		.filter_map(|(key, attributes)| {
-			let key = key as u8;
+			let key = quote::quote! { #path::__private::codec::Compact(#key as u8) };
 			let deprecation_status = parse_deprecation(path, attributes).transpose();
 			deprecation_status.map(|item| item.map(|item| quote::quote! { (#key, #item) }))
 		})
 		.collect::<Result<Vec<TokenStream>>>()?;
-
-	Ok(
-		quote::quote! { #path::__private::scale_info::prelude::collections::BTreeMap::from([#( #children),*]) },
-	)
+	match (parent_deprecation, children.as_slice()) {
+		(None, []) =>
+			Ok(quote::quote! { #path::__private::metadata_ir::DeprecationInfoIR::NotDeprecated }),
+		(None, _) => {
+			let children = quote::quote! { #path::__private::scale_info::prelude::collections::BTreeMap::from([#( #children),*]) };
+			Ok(
+				quote::quote! { #path::__private::metadata_ir::DeprecationInfoIR::PartiallyDeprecated(#children) },
+			)
+		},
+		(Some(depr), []) => Ok(
+			quote::quote! { #path::__private::metadata_ir::DeprecationInfoIR::FullyDeprecated(#depr) },
+		),
+		(Some(_), _) => {
+			let span = parent_attrs
+				.iter()
+				.find(|a| a.path().is_ident("deprecated"))
+				.map(|x| x.span())
+				.expect("this can never fail");
```

Comment:
nit: "because we have found the deprecated attribute; qed"?

Reply:
changed the explanation according to the suggestion